### DISCUSSION TITLE
GNPy 2.3

### DISCUSSION THREAD

**Discussion by jktjkt** [Telecominfraproject/oopt-gnpy#406] [General]
Hello from the virtual OFC'2021. A fresh release of GNPy, a transmission quality estimator for DWDM optical networks, is here.

In this release, we added support for modeling of OpenROADM networks. Example:

```console-session
gnpy-transmission-example \
  -e gnpy/example-data/eqpt_config_openroadm.json \
  gnpy/example-data/Sweden_OpenROADM_example_network.json
```

We have improved our documentation, so that it is hopefully easier to navigate and covers more advanced topics. We've also extended the docs with information targeted at vendors (or others) who are willing to contribute equipment datasheets to GNPy.

ROADMs can now specify different target per-channel launch powers for different directions (degrees). We've also fixed various bugs; please refer to the git changelog for a detailed list.

Internally, we've extended the test coverage of our code base, improved the CI infrastructure, and addressed some technical debt. However, we've grown limited by supporting Python 3.6, so this is the last release which works with that version. The next release will require Python 3.8.

There's much more brewing in our development branches, so I'm already looking forward to the next release. Take a look behind the curtain:

 https://review.gerrithub.io/q/project:Telecominfraproject/oopt-gnpy

<hr /><em>This discussion was created from the release <a href='https://github.com/Telecominfraproject/oopt-gnpy/releases/tag/v2.3'>GNPy 2.3</a>.</em>

**Comment by jktjkt** [comment]
Just a small heads-up: I also tagged a follow-up release, [`v2.3.1`](https://github.com/Telecominfraproject/oopt-gnpy/releases/tag/v2.3.1). There are **no user-facing changes** in that release, and there's no need to update. I had to make that release in order to solve a technical issue with our Continuous Integration. As a side effect, the Docker images are available once again, and the [landing page at pypi](https://pypi.org/project/gnpy/) shows correct information.

**Reply by gv100204** [reply]
Hi!
Thanks for your work, really appreciated!
One small request I have is if it's possible to have a more detailed documentation about APIs and some examples from scratch that are not available in this release.
Thanks

**Reply by jktjkt** [reply]
Thanks, we're working on that. In the meanwhile, we [have some videos](https://github.com/Telecominfraproject/oopt-gnpy/discussions/378#discussioncomment-282301) that might help.
